be/src/olap/rowset/segment_v2/binary_prefix_page.cpp

```diff
@@ -212,13 +213,17 @@ Status BinaryPrefixPageDecoder::next_batch(size_t* n, vectorized::MutableColumnP
     }
     size_t max_fetch = std::min(*n, static_cast<size_t>(_num_values - _cur_pos));
 
+    dst->insert_data((char*)(_current_value.data()), _current_value.size());
     // read and copy values
-    for (size_t i = 0; i < max_fetch; ++i) {
-        dst->insert_data((char*)(_current_value.data()), _current_value.size());
-        _read_next_value();
+    for (size_t i = 0; i < max_fetch - 1; ++i) {
+        RETURN_IF_ERROR(_read_next_value());
         _cur_pos++;
+        dst->insert_data((char*)(_current_value.data()), _current_value.size());
     }
 
+    RETURN_IF_ERROR(_read_next_value());
```

Comment:
If you reached the end of data, should not read next value anymore, or you would return a CorruptionError here.